File: Patterns/Java/Creational/Builder/BuilderVestCap.java
public class BuilderVestCap extends Builder{

    BuilderVestCap(){
        outfit = new Outfit();
    }

    public void buildCoat(){
        outfit.addClothe( new Vest() );
    }


    public void buildHat(){
        outfit.addClothe( new Cap() );
    }

    public Outfit getOutfit(){
        return outfit;
    }

}

File: Patterns/C#/Creational/Builder/BuilderVestCap.cs
public class BuilderVestCap : Builder{

    public BuilderVestCap(){
        outfit = new Outfit();
    }

    public override void buildCoat(){
        outfit.addClothe( new Vest() );
    }

    public override void buildHat(){
        outfit.addClothe( new Cap() );
    }

    public override Outfit getOutfit(){
        return outfit;
    }

}
